src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs

```diff
@@ -1295,6 +1295,215 @@ public void PasteWithSelectionInReadonlyArea()
             AssertCaretVirtualPosition(2, 4);             
         }
 
+        [Fact]
+        public void DeleteLineWithOutSelection()
+        {
+            Submit(
+@"1",
+@"1
+");                                                                                                                        
+            var caret = Window.TextView.Caret;                               
+
+            // DeleteLine with caret in readonly area
+            caret.MoveToPreviousCaretPosition();
+            caret.MoveToPreviousCaretPosition();
+            caret.MoveToPreviousCaretPosition();
+
+            AssertCaretVirtualPosition(1, 1);
+            Task.Run(() => Window.Operations.DeleteLine()).PumpingWait();
```

Comment:
Generally speaking, there are two reasons we might invoke a method on a background thread:
 1. We're testing how it behaves on a background thread.  Those tests have already been written, but feel free to expand them.
 2. It should have been async, but it isn't and we have to use ```PumpingWait``` to determine when it finishes.  We should never need to do this for new methods, because they can be made async.